src/webgpu/gpu_test.ts

```diff
@@ -416,6 +442,72 @@ got [${failedByteActualValues.join(', ')}]`;
     return undefined;
   }
 
+  checkSingleValueBuffer(
+    actual: TypedArrayBufferView,
+    exp: TypedArrayBufferView,
+    tolerance: number | ((i: number) => number) = 0
+  ): string | undefined {
```

Comment:
I'm sure we'll refactor all these helpers eventually anyway, but maybe this could be written on top of checkBufferFn to make it duplicate less?

Reply:
We discussed this online and unfortunately checkBufferFn doesn't generate the error messages itself - the function passed to it does - so it doesn't share a lot of code.

A better solution would be to carefully refactor checkBuffer to take functions that index into the actual and expected arrays, or similar. Per discussion, it sounds like we could do this in a follow-up.